SSM/Abilities/BouncyBacon.java

```diff
@@ -0,0 +1,75 @@
+package SSM.Abilities;
+
+import SSM.GameManagers.DisguiseManager;
+import SSM.GameManagers.Disguises.NetherPigDisguise;
+import SSM.GameManagers.OwnerEvents.OwnerRightClickEvent;
+import SSM.Projectiles.BaconProjectile;
+import SSM.Projectiles.IronHookProjectile;
+import SSM.Projectiles.SulphurProjectile;
+import SSM.Utilities.Utils;
+import net.minecraft.server.v1_8_R3.EnumParticle;
+import org.bukkit.ChatColor;
+import org.bukkit.Material;
+import org.bukkit.Sound;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Item;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.event.player.PlayerPickupItemEvent;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BouncyBacon extends Ability implements OwnerRightClickEvent {
+
+    private float energy_to_use = 0.225f;
+    private long lastTimeUsed;
+    private List<Entity> bouncyBacons = new ArrayList<>();// eh system could be memory leak or something
```

Comment:
You didn't make this static so there won't be memory leaks, but it should be static since other pigs can eat any bouncy bacon